walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,124 @@
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private static final int START_OF_DAY = TimeRange.START_OF_DAY;
+    private static final int END_OF_DAY = TimeRange.END_OF_DAY;
+    private static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+    private static final Comparator<TimeRange> ORDER_BY_START = TimeRange.ORDER_BY_START;
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
+        // Array of queries represenitng times to meet.
+        List<TimeRange> query = new ArrayList<>();
+        
+        int numberOfEvents = events.size();
+        
+        int timeOfStart = START_OF_DAY;
+        
+        int desiredDuration = (int) request.getDuration();
+        
+        //  Event counter used whilst iterating.
+        int eventCounter = 0;
+        
+        int eventStart = 0;
+        int eventEnd = 0;
+
+        // Set of stirngs representing unique employees set to attend this meeting.
+        Set<String> attendees = new HashSet();
+
+        // Collection of string reprenting who is to attend the requedted meeting
+        Collection<String> requestAttendees = request.getAttendees();
+        
+        // Boolean used to indictae whter or not the attendee should be ignored.
+        boolean shouldIgnore = true;
+
+        // Check if duration of event is longer than a whole day and if so return empty.
+        if ((int)request.getDuration() > WHOLE_DAY.duration()) {
+            return query;
+        }
+
+        // Check if there are any events happening at all if so return whole day query.
+        if (numberOfEvents == 0) {
+            // Adds a 24 hour time range to query.
+            query.add(TimeRange.WHOLE_DAY);
+            return query;
+        }
+
+        // Iterating throguh input event collection. 
+        for (Event event : events) {
+
+            // Collection of the pople who are to attend meeting.
+            attendees = event.getAttendees();
+
+            // Ignore event only if attendees of this event are not needed in requested meeting
+            for (String attendee : attendees) {
+                // If this attendee is required
+                if (requestAttendees.contains(attendee)) {
+                    // You shouldn't ignore this attendee. 
+                    shouldIgnore = false;
+                    break;
+                } 
+                // If at the last event.
+                if (eventCounter == numberOfEvents-1) {
+                    // Add time range of current start time to end of day.
+                    query.add(TimeRange.fromStartEnd(timeOfStart, END_OF_DAY, /* inclusive= */ true));
+                }
+            }
+            // if the attendee is mandatory.
```

Comment:
It does represent when an individual attendee is mandatory but it resets to true after checking each individual attendee

Reply:
Is `shouldIgnore` false when at least one attendee at `event` is required? If so then I would update this comment since this variable doesn't refer to a single attendee.

Reply:
Done